src/main/scala/dxWDL/compiler/IR.scala

```diff
@@ -50,13 +50,53 @@ object IR {
                                    tag: Option[Vector[String]])
       extends PatternsRepr
   
+  /** Compile time representation of the dxapp IO spec choices
+    * Choices is an array of choice values, where each value can be raw (a primitive type)
+    * or annotated (a hash with optional 'name' key and required 'value' key). We represent
+    * both types of values as a class with 'name' and 'value' members. When translating to
+    * dxapp.json, we write a raw value if 'name' is None, and an annotated value if not.
+    *  Examples:
+    *   'choices': [
+    *     {
+    *       'name': "yes", 'value': true
+    *     },
+    *     {
+    *       'name': "no", 'value': false
+    *     }
+    *   ]  # => [{'name': "yes", 'value': true}, {'name': "no", 'value': false}]
+    *     
+    *   'choices': [true, false]  # => [true, false]
+    *
+    *   'choices': [{'value': 1}, {'value': 2}]  # => [1, 2]
+  **/
+  sealed abstract class ChoiceRepr
+  final case class ChoiceReprString(value: String) extends ChoiceRepr
+  final case class ChoiceReprInteger(value: Int) extends ChoiceRepr
+  final case class ChoiceReprFloat(value: Double) extends ChoiceRepr
+  final case class ChoiceReprBoolean(value: Boolean) extends ChoiceRepr
+  final case class ChoiceReprFile(value: String, name: Option[String]) extends ChoiceRepr
+
+  sealed abstract class SuggestionRepr
+  sealed case class SuggestionReprString(value: String) extends SuggestionRepr
+  sealed case class SuggestionReprInteger(value: Int) extends SuggestionRepr
+  sealed case class SuggestionReprFloat(value: Double) extends SuggestionRepr
+  sealed case class SuggestionReprBoolean(value: Boolean) extends SuggestionRepr
+  sealed case class SuggestionReprFile(
+    value: Option[String],
+    name: Option[String],
+    project: Option[String],
+    path: Option[String],
+  ) extends SuggestionRepr
+
   // Compile time representaiton of supported parameter_meta section
   // information for the dxapp IO spec.
```

Comment:
Can you add a comment describing that this is from dxapp.json. Something like this:
```
 {
      "name": "sorted_bams",
      "label": "Sorted mappings",
      "help": "A set of coordinate-sorted BAM files to be merged.",
      "class": "array:file",
      "patterns": ["*.bam"]
    },
```
https://documentation.dnanexus.com/developer/apps/advanced-app-tutorial#the-dxapp-json

Reply:
Yep